internal/nvmeof/controller/controllerserver.go

```diff
@@ -399,6 +399,21 @@ func (cs *Server) ControllerModifyVolume(
 	return &csi.ControllerModifyVolumeResponse{}, nil
 }
 
+// ControllerExpandVolume handles volume expansion requests.
+// For now it only updates the capacity in the response as NVMe-oF
+// this must be added because ControllerModifyVolume requires the sidecar csi-resizer. and
+// csi-resizer searches for the capacity ControllerServiceCapability_RPC_EXPAND_VOLUME.
+// In the future, if NVMe-oF gateway supports volume expansion, the logic must be added here.
```

Comment:
This sounds like a shortcoming of the external-resizer sidecar. Only ModifyVolume should be allowed. But there seems to be a required check on volume resize: https://github.com/kubernetes-csi/external-resizer/blob/master/pkg/resizer/csi_resizer.go#L59

Reply:
created https://github.com/kubernetes-csi/external-resizer/pull/545 for that, but a **draft** for now as I have not tested it yet

Reply:
yes,`ControllerExpandVolume` needs to be implemented too, but in a separate PR :smile: